build.sh

```diff
@@ -372,6 +389,38 @@ if (( ${NUMARGS} == 0 )) || hasArg libraft || hasArg docs || hasArg tests || has
         cmake --build  "${LIBRAFT_BUILD_DIR}" ${VERBOSE_FLAG} -j${PARALLEL_LEVEL} --target ${CMAKE_TARGET}
       fi
   fi
+  compile_end=$(date +%s)
+  compile_total=$(( compile_end - compile_start ))
+
+  if [[ "$BUILD_REPORT_METRICS" == "ON" && -f "${LIBRAFT_BUILD_DIR}/.ninja_log" ]]; then
+      if ! rapids-build-metrics-reporter.py 2> /dev/null && [ ! -f rapids-build-metrics-reporter.py ]; then
+          echo "Downloading rapids-build-metrics-reporter.py"
+          curl -sO https://raw.githubusercontent.com/rapidsai/build-metrics-reporter/v1/rapids-build-metrics-reporter.py
+      fi
+
+      echo "Formatting build metrics"
+      MSG=""
+      # get some sccache stats after the compile
+      if [[ "$BUILD_REPORT_INCL_CACHE_STATS" == "ON" && -x "$(command -v sccache)" ]]; then
+          COMPILE_REQUESTS=$(sccache -s | grep "Compile requests \+ [0-9]\+$" | awk '{ print $NF }')
+          CACHE_HITS=$(sccache -s | grep "Cache hits \+ [0-9]\+$" | awk '{ print $NF }')
+          HIT_RATE=$(echo - | awk "{printf \"%.2f\n\", $CACHE_HITS / $COMPILE_REQUESTS * 100}")
+          MSG="${MSG}<br/>cache hit rate ${HIT_RATE} %"
+      fi
+      MSG="${MSG}<br/>parallel setting: $PARALLEL_LEVEL"
+      MSG="${MSG}<br/>parallel build time: $compile_total seconds"
+      if [[ -f "${LIBRAFT_BUILD_DIR}/libraft.so" ]]; then
+          LIBRAFT_FS=$(ls -lh ${LIBRAFT_BUILD_DIR}/libraft.so | awk '{print $5}')
+          MSG="${MSG}<br/>libraft.so size: $LIBRAFT_FS"
+      fi
+      BMR_DIR=${RAPIDS_ARTIFACTS_DIR:-"${LIBRAFT_BUILD_DIR}"}
+      echo "Metrics output dir: [$BMR_DIR]"
+      mkdir -p ${BMR_DIR}
+      MSG_OUTFILE="$(mktemp)"
+      echo "$MSG" > "${MSG_OUTFILE}"
+      PATH=".:$PATH" python rapids-build-metrics-reporter.py ${LIBRAFT_BUILD_DIR}/.ninja_log --fmt html --msg "${MSG_OUTFILE}" > ${BMR_DIR}/ninja_log.html
+      cp ${LIBRAFT_BUILD_DIR}/.ninja_log ${BMR_DIR}/ninja.log
```

Comment:
Can you print a log message where I will be able to find the ninja log html report? Either on disk or the url to the location that CI uploads it to? I want to make sure the build metrics are somewhat discoverable (even if in the log messages in the middle of a conda build...)

Reply:
I just realized that it might also be worth just printing the stats to the console. This might be more readable when this PR to the build-metrics-reporter lands: https://github.com/rapidsai/build-metrics-reporter/pull/1/

Worth considering. Also okay to ignore.

Reply:
How about printing: `The HTML report can be found in ${BMR_DIR}/ninja_log.html. In CI, this report will also be uploaded to the appropriate subdirectory of https://downloads.rapids.ai/ci/raft/.`

Seems less roundabout than:
1. git blame on the line `echo "Metrics output dir: [$BMR_DIR]".`
2. Find it was part of this PR
3. Take the description "This PR will also automatically generate an HTML report in conda-cpp-build CI runs under the task Upload Additional Artifacts"
4. Go to "Upload additional artifacts" in the CI
5. For now: I cannot find that step in the CI for this PR?

I finally found the upload directory by searching through internal communications. Not everybody can do that.

Reply:
`The HTML report can be found in ${BMR_DIR}/ninja_log.html. In CI, this report will also be uploaded to the appropriate subdirectory of https://downloads.rapids.ai/ci/raft/, and the entire URL can be found in "conda-cpp-build" runs under the task "Upload additional artifacts"`

How about this?

Reply:
Haha okay. Now I get it. That link is actually quite easy to find once you know where it is... Yes, your new description helped me find it. I think it will help others as well!